runtime/integration-test/src/xcm/kusama_xcm_transfer.rs

```diff
@@ -0,0 +1,210 @@
+use crate::{
+	setup::*,
+	xcm::{fee_test::*, kusama_test_net::*},
+};
+
+use frame_support::WeakBoundedVec;
+use sp_runtime::traits::{AccountIdConversion, ConstU32};
+use xcm::VersionedXcm;
+use xcm_emulator::TestExt;
+
+pub const MANGATA_ID: u32 = 2110;
+pub const SIBLING_ID: u32 = 2000;
+
+fn mgx_location() -> VersionedMultiLocation {
+	MultiLocation::new(
+		1,
+		X2(
+			Parachain(MANGATA_ID),
+			GeneralKey(WeakBoundedVec::<u8, ConstU32<32>>::force_from(
+				NATIVE_ASSET_ID.encode(),
+				None,
+			)),
+		),
+	)
+	.into()
+}
+
+fn reserve_account(id: u32) -> AccountId {
+	polkadot_parachain::primitives::Sibling::from(id).into_account_truncating()
+}
+
+#[test]
+fn transfer_from_relay_chain() {
+	KusamaRelay::execute_with(|| {
+		assert_ok!(kusama_runtime::XcmPallet::reserve_transfer_assets(
+			kusama_runtime::RuntimeOrigin::signed(ALICE.into()),
+			Box::new(Parachain(MANGATA_ID).into().into()),
+			Box::new(Junction::AccountId32 { id: BOB, network: NetworkId::Any }.into().into()),
+			Box::new((Here, unit(12)).into()),
+			0
+		));
+	});
+
+	Mangata::execute_with(|| {
+		assert_eq!(
+			Tokens::free_balance(RELAY_ASSET_ID, &AccountId::from(BOB)),
+			unit(12) - relay_per_second_as_fee(4)
+		);
+	});
+}
+
+#[test]
+fn transfer_to_relay_chain() {
+	use frame_support::weights::{Weight, WeightToFee as WeightToFeeT};
+	use kusama_runtime_constants::fee::WeightToFee;
+
+	let weight: XcmWeight = 298_368_000;
+	let fee = WeightToFee::weight_to_fee(&Weight::from_ref_time(weight));
+	assert_eq!(103_334_130, fee);
+
+	Mangata::execute_with(|| {
+		assert_ok!(XTokens::transfer(
+			RuntimeOrigin::signed(ALICE.into()),
+			RELAY_ASSET_ID,
+			unit(12),
+			Box::new(
+				MultiLocation::new(
+					1,
+					X1(Junction::AccountId32 { id: BOB, network: NetworkId::Any })
+				)
+				.into()
+			),
+			WeightLimit::Limited(weight)
+		));
+	});
+
+	KusamaRelay::execute_with(|| {
+		assert_eq!(kusama_runtime::Balances::free_balance(&AccountId::from(BOB)), unit(12) - fee);
+	});
+}
+
+#[test]
+fn transfer_asset() {
```

Comment:
are these tests run on ci ?

Reply:
they were run as a part of `cargo test` previously, will check

Reply:
verifed, they are run as part of test